src/http-middlewares/after/log-response.js

```diff
@@ -43,11 +22,11 @@ const prepareRequestLog = (req, resp) => {
     request_type: 'devplatform-outbound',
     request_url: req.url,
     request_method: req.method || 'GET',
-    request_headers: requestHeaders.join('\n'),
+    request_headers: req.headers,
     request_data: req.body,
     request_via_client: true,
     response_status_code: resp.status,
-    response_headers: respHeaders.join('\n'),
+    response_headers: resp.headers,
```

Comment:
So, it turns out the weird check for `headers.forEach` in `prepHeaders` was because `req.headers` gets stored as JSON, but `resp.headers` gets stored as a custom `Headers` object:

![](https://zappy.zapier.com/7DAFA811-4D9C-4368-B657-112DB16EDF88.png)

That being said, there is a `toJSON` method for the `Headers` object:

![](https://zappy.zapier.com/78B8DA19-5C94-465B-8A80-0782DAABEED0.png)

So I think the following should work:

```suggestion
    response_headers: resp.headers.toJSON(),
```

Reply:
ok, so rather than need the folks calling `logger` needing to know to transform their objects, I made logger be more flexible with what it handled. great catch though!